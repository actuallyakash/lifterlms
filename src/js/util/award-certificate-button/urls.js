// WP deps.
import { addQueryArgs } from '@wordpress/url';

// LLMS deps.
import { getAdminUrl } from '@lifterlms/utils';

/**
 * Retrieves the redirect URL for the newly created certificate.
 *
 * @since 6.0.0
 *
 * @param {number} post WP_Post ID.
 * @return {string} The edit post URL for awarded certificate draft.
 */
export function getRedirectUrl( post ) {
	return addQueryArgs(
		`${ getAdminUrl() }/post.php`,
		{
			post,
			action: 'edit',
			newAwardMsg: 1,
		}
	);
}

/**
 * Retrieves the url for the "Start from Scratch" button.
 *
 * @since 6.0.0
 *
 * @param {?number} sid WP_User ID of the selected student.
 * @return {string} The URL.
 */
export function getScratchUrl( sid = null ) {
	const args = {
		post_type: 'llms_my_certificate',
	};

	if ( sid ) {
		args.sid = sid;
	}

	return addQueryArgs(
		`${ getAdminUrl() }/post-new.php`,
		args,
	);
}
