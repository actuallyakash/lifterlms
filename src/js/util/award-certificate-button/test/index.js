import getMessage from '../message';
import { getScratchUrl, getRedirectUrl } from '../urls';

describe( 'AwardCertificateButton', () => {
	describe( 'getMessage', () => {
		const testData = [
			[ 'No student and no template', null, null ],
			[ 'Student and no template', 123, null ],
			[ 'Template and no student', null, 456 ],
			[ 'Template and student', 1, 2 ],
		];
		test.each( testData )( '%s', ( name, studentId, templateId ) => {
			expect( getMessage( studentId, templateId ) ).toMatchSnapshot();
		} );
	} );

	describe( 'urls', () => {
		describe( 'getRedirectUrl', () => {
			test( 'Returns the URL', () => {
				expect( getRedirectUrl( 123 ) ).toMatchSnapshot();
			} );
		} );

		describe( 'getScratchUrl', () => {
			const testData = [
				[ 'With student ID', 123 ],
				[ 'Without student ID', null ],
			];
			test.each( testData )( '%s', ( name, studentId ) => {
				expect( getScratchUrl( studentId ) ).toMatchSnapshot();
			} );
		} );
	} );
} );
