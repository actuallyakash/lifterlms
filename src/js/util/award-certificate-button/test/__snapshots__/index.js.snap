// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwardCertificateButton getMessage No student and no template 1`] = `""`;

exports[`AwardCertificateButton getMessage Student and no template 1`] = `"Create a new certificate award from this template for the selected student."`;

exports[`AwardCertificateButton getMessage Template and no student 1`] = `"Create a new certificate award from the selected template for this student."`;

exports[`AwardCertificateButton getMessage Template and student 1`] = `"Create a new certificate award from the selected template for the selected student."`;

exports[`AwardCertificateButton urls getRedirectUrl Returns the URL 1`] = `"/wp-admin/post.php?post=123&action=edit&newAwardMsg=1"`;

exports[`AwardCertificateButton urls getScratchUrl With student ID 1`] = `"/wp-admin/post-new.php?post_type=llms_my_certificate&sid=123"`;

exports[`AwardCertificateButton urls getScratchUrl Without student ID 1`] = `"/wp-admin/post-new.php?post_type=llms_my_certificate"`;
